pkg/kapp/cmd/core/deps_factory.go

```diff
@@ -58,6 +59,26 @@ func (f *DepsFactoryImpl) DynamicClient() (dynamic.Interface, error) {
 	return clientset, nil
 }
 
+func (f *DepsFactoryImpl) MutedDynamicClient() (dynamic.Interface, error) {
+	config, err := f.configFactory.RESTConfig()
```

Comment:
As the code is almost similar to Dynamic Client, should we have a common function being called from both MutedDynamicClient and DynamicClient by passing the boolean flag for enabling or disabling the warning?